locals.tf

```diff
@@ -16,40 +15,56 @@ locals {
     http_put_response_hop_limit = 2 # Hop limit should be between 2 and 64 for IMDSv2 instance metadata services
   }
 
+  eks_cluster_id     = module.aws_eks.cluster_id
+  cluster_ca_base64  = module.aws_eks.cluster_certificate_authority_data
+  cluster_endpoint   = module.aws_eks.cluster_endpoint
+  vpc_id             = var.vpc_id
+  private_subnet_ids = var.private_subnet_ids
+  public_subnet_ids  = var.public_subnet_ids
+  tags               = module.eks_tags.tags
+
+  enable_workers                    = length(var.self_managed_node_groups) > 0 || length(var.managed_node_groups) > 0 ? 1 : 0
```

Comment:
Should this be boolean?

Reply:
This is used in resource `count` hence setting this to 1 or 0 .  Condition will be much simpler now e.g., `count = var.enable_workers`